src/app/qgisapp.cpp

```diff
@@ -1940,6 +1950,137 @@ QgsMessageBar *QgisApp::visibleMessageBar()
   }
 }
 
+QList<QgsVectorLayerRef> QgisApp::findBrokenWidgetDependencies( QgsVectorLayer *vl )
+{
+  QList<QgsVectorLayerRef> brokenDependencies;
+  // Check for missing layer widget dependencies
+  for ( int i = 0; i < vl->fields().count(); i++ )
+  {
+    std::unique_ptr<QgsEditorWidgetWrapper> ww;
+    ww.reset( QgsGui::editorWidgetRegistry()->create( vl, i, nullptr, nullptr ) );
+    for ( QgsVectorLayerRef &dependency : ww->layerDependencies() )
```

Comment:
Take a const copy before iterating a container with a range-based for loop or donald trump riding a unicorn singing elton john songs will appear every night in front of your window. And believe me it sounds terrible when he sings elton john.

Reply:
It's not that I like Elton John, but `resolveWeakly` is not const.

Reply:
Hmm... So what happens is you get the `layerDependencies`, take a deep copy of it to iterate, then `resolveWeakly()` on each of the items and drop the modified container again, while forwarding the individual items. This kind of works, so maybe you get the same scenario but muted. If you can live with that, I can as well :+1: I think to get rid of unicorn and president as well you'll need to take a copy only of each item in the loop without a detach in the beginning.